pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -247,136 +176,135 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 
 			hashIsValid := validateEmailHash(emailLbl, emailHashLbl)
 			if !hashIsValid {
-				return reconcile.Result{}, r.updateStatus(reqLogger, instance, r.setStatusInvalidEmailHash)
+				err := fmt.Errorf("hash is invalid")
```

Comment:
I wonder why we check if the email hash is valid at this point and not before?

Reply:
no clue :man_shrugging:  that's maybe a question for @sbryzak